auth_backends/backchannel_logout.py

```diff
@@ -0,0 +1,108 @@
+
+        if self.id_token_issuer() != logout_token.get('iss'):
+            raise AuthTokenError(self, 'Incorrect logout_token: iss')
+
+        # Verify the token was issued in the last ID_TOKEN_MAX_AGE seconds
+        iat_leeway = self.setting('ID_TOKEN_MAX_AGE', self.ID_TOKEN_MAX_AGE)
+        if utc_timestamp > logout_token.get('iat') + iat_leeway:
+            raise AuthTokenError(self, 'Incorrect logout_token: iat')
+
+        if not logout_token.get('sub'):
+            raise AuthTokenError(self, 'Incorrect logout_token: sub')
+
+        events = logout_token.get('events')
+        try:
+            if not events or 'http://schemas.openid.net/event/backchannel-logout' not in events.keys():
+                raise AuthTokenError(self, 'Incorrect logout_token: events')
+        except AttributeError:
+            raise AuthTokenError(self, 'Incorrect logout_token: events')
+
+        if logout_token.get('nonce'):
+            raise AuthTokenError(self, 'Incorrect logout_token: nonce')
+
+    def validate_and_return_logout_token(self, logout_token):
+        """
+        Validates the logout_token according to the steps at
+        https://openid.net/specs/openid-connect-backchannel-1_0.html#Validation.
+        """
+        client_id, client_secret = self.get_key_and_secret()
+
+        try:
+            key = self.find_valid_key(logout_token)
+        except ValueError:
+            raise AuthTokenError(self, 'Incorrect logout_token: key missing')
+
+        if not key:
+            raise AuthTokenError(self, 'Signature verification failed')
+
+        alg = key['alg']
+        rsa_key = jwk.construct(key)
+
+        try:
+            claims = jwt.decode(
+                logout_token,
+                rsa_key.to_pem().decode('utf-8'),
+                algorithms=[alg],
+                audience=client_id,
+                options=self.JWT_DECODE_OPTIONS,
+            )
+        except ExpiredSignatureError:
+            raise AuthTokenError(self, 'Signature has expired')
+        except JWTClaimsError as error:
+            raise AuthTokenError(self, str(error))
+        except JWTError:
+            raise AuthTokenError(self, 'Invalid signature')
+
+        self.validate_logout_claims(claims)
+
+        return claims
+
+    def backchannel_logout(self, *args, **kwargs):
+        post_data = self.strategy.request_post()
+
+        logout_token = post_data.get('logout_token')
+        if not logout_token:
+            raise AuthException(self, 'Log out token missing')
+
+        claims = self.validate_and_return_logout_token(logout_token)
+
+        social_auth = self.strategy.storage.user.get_social_auth(
+            self.name,
+            claims.get('sub'),
+        )
+        if not social_auth:
+            raise AuthException(self, 'User not authenticated with this backend')
+
+        # Notice: The following is a Django specific session deletion
+        User = get_user_model()  # noqa
```

Comment:
If this noqa is because of the sonar rule for lowercase variable names, we can discuss turning the rule off for Django.

Reply:
Yes it is. No, I don't see the need to turn it off entirely. Couple of noqa here and there hurt no-one.

Reply:
I don't like these directives in code that tune the behavior of some tool either. The tool should be configured, or coding style should be changed.

Reply:
This specific rule: The pattern is pretty common in Django, especially in migrations (which of course could be handled as a path-based exclusion. But I'm thinking the benefits for this rule are pretty minimal in Django code.

Reply:
Yep, it's common. And I think it's ok. In that sense, perhaps the rule should just be disabled in Sonar?